Encapsulate position setpoint and loiter setpoint handling for FW position controller

Jaeyoung-Lim opened the pull request:
**Describe problem solved by this pull request**
The FW position controller logic is quite complicated, with different exception handled in multiple points.

This is an attempt to encapsulate position setpoint handling and loiter setpoint handling, so that it is easier to understand the logic behind how the waypoints are handled.

Most of the logic is currently based on curr, prev, and next waypoint in global coordinates. While we want to switch to trajectory setpoints, or at least local coordinates I think a gradual cleanup would benefit on moving this direction

**Test data / coverage**
Tested various waypoints in PX4 SITL Gazebo

sfuhrer commented on src/modules/fw_pos_control_l1/FixedwingPositionControl.cpp:
```diff
@@ -1235,6 +1007,376 @@ FixedwingPositionControl::control_position(const hrt_abstime &now, const Vector2
+		const Vector2f &ground_speed, const position_setpoint_s &pos_sp_prev, const position_setpoint_s &pos_sp_curr,
+		const position_setpoint_s &pos_sp_next)
+{
+	Vector2f nav_speed_2d{ground_speed};
```
Did you on purpose replace nav_speed_2d by simple groundspeed in navigate_waypoints? Currently this variable is not consumed anywhere if I'm not mistaken.

Jaeyoung-Lim replied:
@sfuhrer

Jaeyoung-Lim replied:
@sfuhrer Ah, I was thinking maybe we should handle that inside the l1 controller since it is l1 specific. But I think it is already better!

sfuhrer reviewed (commented):
Thanks @Jaeyoung-Lim ! This it seems like a good first step into the right direction. 
I'm sure we find some other low hanging fruits to clean up the FW Position controller, I'll also dig around in it the next days. Especially those FW takeoff/landing customization are all over the place, hopefully we can group them better.

Jaeyoung-Lim commented:
Rebased due to conflicts caused by https://github.com/PX4/PX4-Autopilot/pull/18350

Jaeyoung-Lim commented:
Rebased due to conflicts created by https://github.com/PX4/PX4-Autopilot/pull/18381

Jaeyoung-Lim commented:
@dagar I have moved the mode handling logic outside of the `control_position`, and renamed `control_position` as `control_auto` since it is now more of a function handling different auto setpoints

dagar commented:
> @dagar I have moved the mode handling logic outside of the `control_position`, and renamed `control_position` as `control_auto` since it is now more of a function handling different auto setpoints

Make sense to me.

Jaeyoung-Lim commented:
@RomanBapst Great to hear that! credit also goes to @sfuhrer Thanks! merging!
